tests/core/test_pod_security_policies.py

```diff
@@ -74,57 +89,56 @@ def test_service_accounts_have_role_binding(admin_mc, request):
     wait_for(lambda: service_account_has_role_binding(rbac, pspt), timeout=30)
 
 
-def test_pod_security_policy_template_del(admin_mc, request, remove_resource):
+def test_pod_security_policy_template_del(admin_mc, admin_pc, remove_resource):
     """ Test for pod security policy template binding correctly
     ref https://github.com/rancher/rancher/issues/15728
     ref https://localhost:8443/v3/podsecuritypolicytemplates
     """
     api_client = admin_mc.client
     # these create a mock pspts... not valid for real psp's
 
-    pspt_proj = api_client.create_pod_security_policy_template("pspt2")
+    pspt_proj = create_pspt(api_client)
     # add a finalizer to delete the pspt
     remove_resource(pspt_proj)
 
-    #  create a project in order to establish bindings
-    def create_project():
-        p = api_client.create_project(name="test-unrestrict-proj",
-                                      clusterId="local")
-        p = api_client.wait_success(p)
-        # In case something goes badly, add a finalizer to
-        # delete the project as the admin
-        remove_resource(p)
-
-        p.setpodsecuritypolicytemplate(
-            podSecurityPolicyTemplateId=pspt_proj.id)
-
-        return p
-
-    proj = create_project()
-    # wait to use proj object until done transitioning
+    #  creates a project and handles cleanup
+    proj = admin_pc.project
+    proj.setpodsecuritypolicytemplate(podSecurityPolicyTemplateId=pspt_proj.id)
     proj = api_client.wait_success(proj)
+
     # Check that project was created successfully with pspt
     assert proj.state == 'active'
     assert proj.podSecurityPolicyTemplateId == pspt_proj.id
 
+    def check_psptpb():
+        proj_obj = proj.podSecurityPolicyTemplateProjectBindings()
+        data_list = proj_obj.data
+        for data in data_list:
+            if (data.targetProjectId == proj.id and
+               data.podSecurityPolicyTemplateId == pspt_proj.id):
+                return True
+
+        return False
+
+    wait_for(check_psptpb)
```

Comment:
Use the `fail_handler` here to clarify why the timeout fails as there is not an assert after the wait_for.